packages/flutter_tools/lib/src/android/android_sdk.dart

```diff
@@ -320,6 +320,43 @@ class AndroidSdk {
 
   String? getAvdManagerPath() => getCmdlineToolsPath(globals.platform.isWindows ? 'avdmanager.bat' : 'avdmanager');
 
+  String? getNdkCompilerPath(String binaryName) {
+    final Directory ndk = directory.childDirectory('ndk');
+    if (!ndk.existsSync()) {
+      return null;
+    }
+    final List<Directory> ndkVersions =
```

Comment:
Might be nice to have a comment explaining that we select the first version and first host architecture that has the file we are looking for and _why_ that is reasonable.

Would a better algorithm try and select the _latest_ version of the NDK?